### PR TITLE
Add OpenSearch 2.0 to the test matrix and dependency

### LINKED ISSUE #159: Release version 2.0.0 for Java client
Release the client version 2.0.0. Before releasing, the following items need to be done:

- [x] Use OpenSearch 2.0 in [gradle](https://github.com/opensearch-project/opensearch-java/blame/e14fb59fa7582521f84d5283e39c0aeb86fb3a45/java-client/build.gradle.kts#L126) dependencies in place of the snapshot version. 
- [x] Add OpenSearch 2.0 to the integration [test matrix](https://github.com/opensearch-project/opensearch-java/blob/main/.github/workflows/test-integration.yml#L17).
- [x] Bump the client version to 2.0.0.
- [x] Cut a 2.0 branch for the client.

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha Shah <vachshah@amazon.com>

### Description
Adding OpenSearch 2.0 to the test matrix. Also updating the rest-client dependency to 2.0.

*Note*: The artifact is not yet available on maven so leaving this PR in a draft state for now.

### Issues Resolved
#159 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
The artifacts should be available now: https://repo1.maven.org/maven2/org/opensearch/opensearch/2.0.0/

**Comment by VachaShah** [comment]
> The artifacts should be available now: https://repo1.maven.org/maven2/org/opensearch/opensearch/2.0.0/

Thank you @reta ! I will re-run the workflows and publish the PR.

File: java-client/build.gradle.kts [hunk 1]
```diff
@@ -123,7 +123,7 @@ val integrationTest = task<Test>("integrationTest") {
 
 dependencies {
 
-    val opensearchVersion = "2.0.0-alpha1-SNAPSHOT"
+    val opensearchVersion = "2.0.0"
     val jacksonVersion = "2.12.6"
```
Comment (reta) [inline java-client/build.gradle.kts:127]: Please update Jakson as well [1]:

```
val jacksonVersion = "2.13.2"
val jacksonDatabindVersion = "2.13.2.2"
```

[1] https://github.com/opensearch-project/OpenSearch/blob/2.0/buildSrc/version.properties#L12

Comment (VachaShah) [reply java-client/build.gradle.kts]: Sure! Updated the jackson versions.

**Review by reta — APPROVED** [review]

**Review by CEHENKLE — APPROVED** [review]

File: java-client/build.gradle.kts (hunk 1 above)
Comment (reta) [reply java-client/build.gradle.kts]: :+1:

Comment (saratvemulapalli) [reply java-client/build.gradle.kts]: @reta I recall you had a suggestion(via gradle) to import the versions from opensearch. 
Is it something we could use here?

**Review by saratvemulapalli — APPROVED** [review]

File: java-client/build.gradle.kts (hunk 1 above)
Comment (reta) [reply java-client/build.gradle.kts]: @saratvemulapalli that would be great but we haven't done any work towards that yet (afaik)

Comment (saratvemulapalli) [reply java-client/build.gradle.kts]: Opened up an issue: https://github.com/opensearch-project/opensearch-java/issues/161
Not sure if there is one on OpenSearch, if not I'll open one.

**Pull request merged**